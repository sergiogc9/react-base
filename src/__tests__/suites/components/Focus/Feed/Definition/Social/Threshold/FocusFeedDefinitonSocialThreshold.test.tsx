import React from 'react';
import { shallow } from "enzyme";

import FocusFeedDefinitonSocialThreshold from '@src/components/Focus/Feed/Definition/Social/Threshold/FocusFeedDefinitonSocialThreshold';

describe("Social definition threshold tests", () => {
	let wrapper: any;
	let onCheckToggled: any;
	let onValueChanged: any;

	beforeEach(() => {
		onCheckToggled = jest.fn();
		onValueChanged = jest.fn();

		wrapper = shallow(
			<FocusFeedDefinitonSocialThreshold
				channel={30}
				value={400}
				error={false}
				errorMessage=''
				disabled={false}
				checked={false}
				onCheckToggled={onCheckToggled}
				onValueChanged={onValueChanged}
			/>
		);
	});

	it("Test component without checked and enabled", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test threshold when checked", () => {
		expect(wrapper.find('.threshold-value-container').exists()).toEqual(false);
		expect(wrapper.find('.threshold-checkbox.checkbox-green').exists()).toEqual(false);
		wrapper.setProps({checked: true});
		expect(wrapper.find('.threshold-value-container').exists()).toEqual(true);
		expect(wrapper.find('.threshold-checkbox.checkbox-green').exists()).toEqual(true);
	});

	it("Test threshold when disabled", () => {
		wrapper.setProps({disabled: true, checked: true});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test threshold when error", () => {
		wrapper.setProps({ error: true, errorMessage: 'Lorem ipsum', checked: true});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test onCheckedToggle call", () => {
		const checkbox = wrapper.find("SelectionControl").at(0);
		checkbox.props().onChange(true);
		expect(onCheckToggled).toHaveBeenCalledWith(true);
	});

	it("Test onValueChanged call", () => {
		wrapper.setProps({checked: true});
		const input =  wrapper.find("TextField").at(0);
		input.props().onChange('');
		expect(onValueChanged).toHaveBeenCalledWith('');
		input.props().onChange('100');
		expect(onValueChanged).toHaveBeenCalledWith(100);
		input.props().onChange('-4');
		expect(onValueChanged).toHaveBeenCalledWith(0);
	});
});
