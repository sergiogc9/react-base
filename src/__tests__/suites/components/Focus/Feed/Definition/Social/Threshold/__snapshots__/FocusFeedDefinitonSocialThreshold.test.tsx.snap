// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social definition threshold tests Test component without checked and enabled 1`] = `"<div class=\\"social-definition-threshold-container \\"><div class=\\"md-selection-control-container threshold-checkbox \\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialThreshold30\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialThreshold30\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialThreshold30\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-label=\\"focusFeedDefinitionSocialThreshold30\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><span class=\\"threshold-name\\">channel.30</span></div>"`;

exports[`Social definition threshold tests Test threshold when disabled 1`] = `"<div class=\\"social-definition-threshold-container disabled-threshold \\"><div class=\\"md-selection-control-container threshold-checkbox checkbox-green\\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialThreshold30\\" checked=\\"\\" disabled=\\"\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialThreshold30\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialThreshold30\\" class=\\"md-selection-control-label md-text--disabled\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-label=\\"focusFeedDefinitionSocialThreshold30\\" class=\\"md-fake-btn md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--disabled\\" disabled=\\"\\" aria-pressed=\\"false\\"><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><span class=\\"threshold-name\\">channel.30</span><span class=\\"threshold-value-container\\">feed.form.threshold.accounts_minimum<div class=\\"md-text-field-container md-inline-block md-text-field-container--disabled md-text-field-container--input threshold-input-value\\"><input type=\\"number\\" value=\\"400\\" min=\\"0\\" id=\\"focusFeedDefinitionSocialThreshold30Value\\" disabled=\\"\\" class=\\"md-text-field md-text-field--margin md-text--disabled\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"/></div>feed.form.threshold.followers, feed.form.threshold.focusing_posts<span class=\\"threshold-estimation\\">51.57%</span>feed.form.threshold.posts</span></div>"`;

exports[`Social definition threshold tests Test threshold when error 1`] = `"<div class=\\"social-definition-threshold-container \\"><div class=\\"md-selection-control-container threshold-checkbox checkbox-green\\"><input type=\\"checkbox\\" id=\\"focusFeedDefinitionSocialThreshold30\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"focusFeedDefinitionSocialThreshold30\\" aria-hidden=\\"true\\"/><label for=\\"focusFeedDefinitionSocialThreshold30\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-label=\\"focusFeedDefinitionSocialThreshold30\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><span class=\\"threshold-name\\">channel.30</span><span class=\\"threshold-value-container\\">feed.form.threshold.accounts_minimum<div class=\\"md-text-field-container md-inline-block md-text-field-container--input threshold-input-value\\"><input type=\\"number\\" value=\\"400\\" min=\\"0\\" id=\\"focusFeedDefinitionSocialThreshold30Value\\" class=\\"md-text-field md-text-field--margin md-text\\"/><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--text-field-error\\"/><div class=\\"md-text-field-message-container md-full-width md-text--error\\"><div aria-hidden=\\"false\\" class=\\"md-text-field-message md-text-field-message--active\\">Lorem ipsum</div></div></div>feed.form.threshold.followers, feed.form.threshold.focusing_posts<span class=\\"threshold-estimation\\">51.57%</span>feed.form.threshold.posts</span></div>"`;
